backend/Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM golang:1.14.7
+COPY . /app
+WORKDIR /app
+
+RUN go mod download \
+&& go mod verify \
+&& go build -o build/app main.go
```

Comment:
Let's separate them into separate `RUN` command so that we leverage Docker layer caching to avoid re downloading all dependencies when `go build` command changes.